source/GameWindow.cpp

```diff
@@ -119,6 +119,7 @@ bool GameWindow::Init()
 		flags |= SDL_WINDOW_FULLSCREEN_DESKTOP;
 	else if(Preferences::Has("maximized"))
 		flags |= SDL_WINDOW_MAXIMIZED;
+	Preferences::SetScreenModeIndex(Preferences::Has("fullscreen"));
```

Comment:
Instead of setting the preference here, you can update the screen mode index inside `Preferences::Load` when the preference is read from file.